splunk/src/main/java/org/apache/calcite/adapter/splunk/SplunkPushDownRule.java

```diff
@@ -243,6 +243,7 @@ protected RelNode appendSearchString(
           field = topFields.get(i);
         }
         newFields.add(field);
+        i++;
       }
```

Comment:
Could you please add a test?

Reply:
And change the commit message to the problem being fixed, not the code change you made to fix it.

Reply:
I will update test cases and commit message.